app/models/manageiq/providers/azure/cloud_manager/vm.rb

```diff
@@ -1,7 +1,9 @@
 class ManageIQ::Providers::Azure::CloudManager::Vm < ManageIQ::Providers::CloudManager::Vm
+  include_concern 'Operations'
+
   def provider_object(connection = nil)
     connection ||= ext_management_system.connect
-    connection.instances[ems_ref]
+    connection
```

Comment:
I'm not sure I understand the gem design, so that could be part of it.  I would expect an Azure::ArmRest::Vm object that exposes all the properties for a Vm.  I would expect class methods for finding different ones based on an id, or methods, like `.all` to get all VMs.  I would expect instance methods, like `.start` to start the VM.

So, the service object feels like that interface, or am I wrong?  What is the difference between a service object and an "actual vm"